robottelo/ui/hostcollection.py

```diff
@@ -130,3 +130,72 @@ def add_host(self, name, host_name):
+                                    action_type='package',
+                                    org_name=None,
+                                    timeout=120):
+        """Execute remote package action on a host collection
+
+        :param name: hostcollection name to remotely execute package action on
+        :param action_name: remote action to execute. Can be one of 4:
+            'update_all', 'install', 'update', 'remove'
+        :param action_value: Package or package group name to remotely
+            install/upgrade/remove (depending on `action_name`)
+        :param action_via: the way to perform the action. Can be one of 3:
+            'katello_agent', 'remote_execution', 'remote_execution_custom'
+        :param action_type: type of package. can be one of 2:
+            'package' or 'package_group'
+        :param org_name: The name of the organization for context, it is an
+            optional param since org can be previously selected on session
+        :param timeout: Timeout in seconds for remote action task to finish
+        :raise: UIError if remote task finished by timeout
+
+        :return: Returns a string containing task status
+        """
+
+        if action_name != 'update_all' and not action_value:
+            raise AttributeError('package or group name is required')
+
+        if org_name:
+            Navigator(self.browser).go_to_select_org(org_name)
+
+        self.click(self.search(name))
+        self.click(tab_locators['hostcollection.collection_actions'])
+        self.click(locators['hostcollection.collection_actions.packages'])
+
+        if action_value:
+            type_loc = "contenthost.bulk_actions.{0}_type".format(action_type)
+            self.click(locators[type_loc])
+            self.assign_value(
+                locators['contenthost.bulk_actions.package_name_input'],
+                action_value
+            )
+
+        # dynamically get locators for dropdown button and action link
+        # based in action_name and action_via
+        action_loc_key = "contenthost.bulk_actions.{action}_via_{via}".format(
+            action=action_name, via=action_via)
+        action_link = locators[action_loc_key]
+
+        # dropdown locator built here instead of locators module
+        # because it is dynamically named
+        dropdown = "/../../../button[contains(@data-toggle,'dropdown')]"
+        action_dropdown_locator = (action_link[0], action_link[1] + dropdown)
+
+        self.click(action_dropdown_locator)  # because element must be visible
+        self.click(action_link)
+
+        result = self.wait_until_element(
+            locators['contenthost.bulk_actions.remote_action_finished'],
+            timeout=timeout,
+        )  # this alert box fades after 2 seconds
+
+        if result is None:
+            raise UIError('Timeout waiting for package action to finish')
+
+        return result.get_attribute('bst-alert')  # u'success|warning|error'
```

Comment:
@abalakh I think it makes sense to wait_until_element with **bst-alert="success"** because this element is available on DOM only after the action button is clicked, if there is something wrong in performing the action that element will never shown in the page and will end in UIError